mesheryctl/internal/cli/root/system/check.go

```diff
@@ -276,7 +277,15 @@ func (hc *HealthChecker) runKubernetesAPIHealthCheck() error {
 			return nil
 		}
 		// else we're supposed to grab the error
-		return errors.New("ctlK8sClient1000: !! cannot initialize a Kubernetes client. See https://docs.meshery.io/reference/error-codes")
+		sdescription := []string{"The Kubernetes cluster is not accessible."}
+		ldescription := []string{fmt.Sprintf("Your %s context is configured to run Meshery on Kubernetes using the %s token", hc.mctlCfg.CurrentContext, hc.context.Token),
+			"The Kubernetes cluster is not accessible",
+			"Please confirm that your cluster is available and that the token is valid",
+			"See https://docs.meshery.io/installation/quick-start for additional instructions",
+		}
+		probablecause := []string{"Kubernetes cluster is unavailable and that the token is invalid"}
+		remedy := []string{"Please confirm that your cluster is available and that the token is valid. See https://docs.meshery.io/installation/quick-start for additional instructions"}
+		return meshkiterrors.New(ErrK8sQueryCode, meshkiterrors.Critical, sdescription, ldescription, probablecause, remedy)
```

Comment:
Can you move the logic to separate function?

Reply:
Also it would be much helpful if you could do this to the all errors present in file.

Reply:
sure I'll move it into a separate function inside [this file ](https://github.com/navendu-pottekkat/meshery/blob/ab49fb36ea7f98db78949d7eb2a75973a156dfc1/mesheryctl/internal/cli/root/system/error.go#L1)